### PR TITLE
MET-4669 Download records from error report

### PR DESCRIPTION
**Pull request by JoanaCMS**

### CODE REVIEW

File: metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/dao/DataEvolutionUtils.java [hunk 1]
```diff
@@ -306,8 +306,13 @@ public List<Pair<ExecutablePlugin, WorkflowExecution>> compileVersionEvolution(M
...
   }
 
-  // Not private because of unit tests.
-  Pair<MetisPlugin, WorkflowExecution> getPreviousExecutionAndPlugin(MetisPlugin plugin, String datasetId) {
+  /**
+   *
+   * @param plugin
+   * @param datasetId
+   * @return
+   */
```
Comment (jochen-vermeulen) [inline metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/dao/DataEvolutionUtils.java:314]: Best to fill this javadoc.

File: metis-core/metis-core-rest/src/main/java/eu/europeana/metis/core/rest/ProxiesController.java [hunk 2]
```diff
@@ -317,4 +317,41 @@ public RecordsResponse getListOfFileContentsFromPluginExecution(
...
+   * @param authorization the authorization header with the access token
+   * @param workflowExecutionId the execution identifier of the workflow
+   * @param pluginType the {@link ExecutablePluginType} that is to be located inside the workflow
+   * @param ecloudIds the list of ecloud IDs of the records we wish to obtain
+   * @return the list of records from the external resource matching the input ID list. If no record
+   * with the matching ID was found in the given workflow step, no entry for this record will appear
+   * in the result list.
+   * @throws GenericMetisException can be one of:
+   * <ul>
+   * <li>{@link eu.europeana.cloud.service.mcs.exception.MCSException} if an error occurred while
```
Comment (jochen-vermeulen) [inline metis-core/metis-core-rest/src/main/java/eu/europeana/metis/core/rest/ProxiesController.java:334]: Should be `eu.europeana.metis.exception.ExternalTaskException`.
(in develop this should have already been changed for the other methods in this class)

File: metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/ProxiesService.java [hunk 3]
```diff
@@ -395,6 +394,52 @@ public RecordsResponse getListOfFileContentsFromPluginExecution(MetisUserView me
...
+   * of the given {@link PluginType}.
+   *
+   * @param metisUserView the user wishing to perform this operation
+   * @param workflowExecutionId the execution identifier of the workflow
+   * @param pluginType the {@link ExecutablePluginType} that is to be located inside the workflow
+   * @param ecloudIds the list of ecloud IDs of the records we wish to obtain
+   * @return the list of records from the external resource
+   * @throws GenericMetisException can be one of:
+   * <ul>
+   * <li>{@link MCSException} if an error occurred while retrieving the records from the external
```
Comment (jochen-vermeulen) [inline metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/ProxiesService.java:408]: Should be `eu.europeana.metis.exception.ExternalTaskException`.
(in develop this should have already been changed for the other methods in this class)

File: metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/ProxiesService.java [hunk 4]
```diff
@@ -395,6 +394,52 @@ public RecordsResponse getListOfFileContentsFromPluginExecution(MetisUserView me
...
+            metisUserView, workflowExecutionId, pluginType);
+    if (executionAndPlugin == null) {
+      throw new NoWorkflowExecutionFoundException(String
+              .format("No executable plugin of type %s found for workflowExecution with id: %s",
+                      pluginType.name(), workflowExecutionId));
+    }
+
+    Pair<MetisPlugin, WorkflowExecution> predecessorPlugin =
+            dataEvolutionUtils.getPreviousExecutionAndPlugin(executionAndPlugin.getRight(), executionAndPlugin.getLeft().getDatasetId());
+    AbstractExecutablePlugin predecessorExecutablePlugin = (AbstractExecutablePlugin) predecessorPlugin.getLeft();
```
Comment (jochen-vermeulen) [inline metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/ProxiesService.java:431]: It is possible that there is no predecessor plugin. Perhaps check for `predecessorPlugin==null` and throw a similar exception as the one just above.

File: metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/dao/DataEvolutionUtils.java (hunk 1 above)
Comment (JoanaCMS) [reply metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/dao/DataEvolutionUtils.java:314]: Oops! Done

File: metis-core/metis-core-rest/src/main/java/eu/europeana/metis/core/rest/ProxiesController.java (hunk 2 above)
Comment (JoanaCMS) [reply metis-core/metis-core-rest/src/main/java/eu/europeana/metis/core/rest/ProxiesController.java:334]: Updated

File: metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/ProxiesService.java (hunk 3 above)
Comment (JoanaCMS) [reply metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/ProxiesService.java:408]: Updated

File: metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/ProxiesService.java (hunk 4 above)
Comment (JoanaCMS) [reply metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/ProxiesService.java:431]: Added

**Comment by JoanaCMS** [comment]
This PR was merged into develop manually

**Pull request closed without merging**